src/main/java/org/sefglobal/scholarx/controller/ProgramController.java

```diff
@@ -53,4 +56,38 @@ public Mentor applyAsMentor(@PathVariable long id,
         long profileId = 1; // TODO: Get the profileId from headers
         return programService.applyAsMentor(id, profileId, mentor);
     }
+
+    @GetMapping("/{id}/mentor")
+    @ResponseStatus(HttpStatus.OK)
+    public Mentor getLoggedInMentor(@PathVariable long id) throws ResourceNotFoundException {
+        long profileId = 1;  // TODO: Get the profileId from headers
+        return programService.getLoggedInMentor(id, profileId);
+    }
+
+    @GetMapping("/{id}/mentees")
+    @ResponseStatus(HttpStatus.OK)
+    public List<Mentee> getMenteesOfAMentor(@PathVariable long id)
+            throws ResourceNotFoundException {
+        long profileId = 1;  // TODO: Get the profileId from headers
+        return programService.getAllMenteesOfAMentor(profileId, id);
+    }
+
+    @PutMapping("/{id}/application")
+    @ResponseStatus(HttpStatus.OK)
+    public Mentor updateApplicationAndPrerequisites(@PathVariable long id,
+                                                    @Valid @RequestBody Mentor mentor)
+            throws ResourceNotFoundException {
+        long profileId = 1;  // TODO: Get the profileId from headers
+        return programService.updateApplicationAndPrerequisites(profileId, id, mentor);
+    }
+
+    @PutMapping("/{id}/mentees/{menteeId}/state")
```

Comment:
Path of this API is `mentees/{id]/state`

Reply:
How do we validate the mentor without a `mentorId` or a `programId`? 

_Find the mentee by id --> Get the mentor of mentee --> Get the profile id of mentor --> Check whether it matches with the profile id taken from header_

Is this method good?

Reply:
No in the frontend we will be loading only the related mentees of a mentor so no need to check if mentee is related to the a `profileId` again.
So,
Find the mentee by id --> Validate the mentee state(pending) --> Approve or reject --> Return updated mentee